ethereum/contracts/pyth/Pyth.sol

```diff
@@ -211,34 +211,23 @@ abstract contract Pyth is PythGetters, PythSetters, AbstractPyth {
     }
 
     function queryPriceFeed(bytes32 id) public view override returns (PythStructs.PriceFeed memory priceFeed){
-
         // Look up the latest price info for the given ID
         PythInternalStructs.PriceInfo memory info = latestPriceInfo(id);
         require(info.priceFeed.id != 0, "no price feed found for the given price id");
 
-        // Check that there is not a significant difference between this chain's time
-        // and the price publish time.
-        if (info.priceFeed.status == PythStructs.PriceStatus.TRADING && 
-            absDiff(block.timestamp, info.priceFeed.publishTime) > validTimePeriodSeconds()) {
-            info.priceFeed.status = PythStructs.PriceStatus.UNKNOWN;
-            // getLatestAvailablePrice* gets prevPrice when status is
-            // unknown. So, now that status is being set to unknown,
-            // we should move the current price to the previous
-            // price to ensure getLatestAvailablePrice* works
-            // as intended.
-            info.priceFeed.prevPrice = info.priceFeed.price;
-            info.priceFeed.prevConf = info.priceFeed.conf;
-            info.priceFeed.prevPublishTime = info.priceFeed.publishTime;
-        }
-
         return info.priceFeed;
```

Comment:
I wonder if we should add `getCurrentPriceNoOlderThan(bytes32 id, uint64 maxAge)` which fails if the cached time is older than `maxAge`.

Reply:
We already have that, `getLatestAvailablePriceWithinDuration`:

https://github.com/pyth-network/pyth-sdk-solidity/blob/main/IPyth.sol#L66

Probably your name looks better though 😅 that was what I and Jayant could think of 😬